mail_outbound_static/models/ir_mail_server.py

```diff
@@ -11,27 +15,62 @@ class IrMailServer(models.Model):
     smtp_from = fields.Char(
         string="Email From", help="Set this in order to email from a specific address."
     )
+    domain_whitelist = fields.Char(
+        help="Allowed domains that can use this SMTP without changing the smtp_from"
```

Comment:
Perhaps we should also add information on the help that this domain will be prefered if not domain specified on the outgoing email